### PR TITLE
Ctk buffer supernova fix

### PR DESCRIPTION
**Pull request by dyfer**
Hi @joshpar 
Supernova fails when it receives buffer messages with number of frames being floats. This has been discussed: https://github.com/supercollider/supercollider/issues/1827 
I don't think there's a consensus how to proceed, but for the time being, the Buffer class is sending buffer messages with numFrames being an integer. I made the relevant changes for CtkBuffer so it does the same. I also updated methods to .asInteger (as opposed to .asInt) for consistency, I hope that's OK.
Marcin
PS. sorry it's messy between the two PRs

### CODE REVIEW

**Comment by dyfer** [comment]
Actually... it seems the plotting still doesn't work all the time, there's some timing sensitivity there. I'll report back once fixed.

File: Ctk classes/CTK.sc [hunk 1]
```diff
@@ -2100,30 +2100,50 @@ CtkBuffer : CtkObj {
...
+						{ //in all other cases
+							syncUsingDoneMsg = false;
+						}
+					);
+					if(syncUsingDoneMsg, {
+						server.sendBundle(latency, bund);
+						cond.wait;
+					}, {
+						server.sendBundle(latency, bund);
+						latency.wait;
```
Comment (dyfer) [inline Ctk classes/CTK.sc]: The original code had server.sync first, then latency.wait; I think this should be other way around, so we wait for latency (which should be how much time server waits with the bundle anyway) and then we ask to sync... is that correct?

**Comment by dyfer** [comment]
This works now.
I changed the sync mechanism to rely on server individual messages after each action, instead of the general server sync method. I hope this looks OK.
I also changed the order of the old sync method, see comment above.

**Comment by joshpar** [comment]
Can you please test on scsynth as well?
Thanks!

**Comment by dyfer** [comment]
These were/are tested on both scsynth and supernova.
However, looking more into it, it did not look as clean when performing multiple operations in a row. I think that the best solution is to send bundles using server.sync(cond, bundles, latency), which seem to work best and takes care of timing indeterminacies that were (I believe) in the root of the problem I'm trying to solve. So this latest commit is the cleanest, and also adds the same sync logic to the .load method. I believe now it IS ready for merging. Sorry for the back and forth!

**Pull request merged**